### PR TITLE
revive pr 237, Added geolocation support for whatsmyip.py using "ip-api.com"

### PR DESCRIPTION
**Pull request by vicyap**
original pr: #237 

credit goes to @neutronst4r for submitting the original pr

Additionally, I have removed the `color_good_online_ip_mode` config https://github.com/ultrabug/py3status/pull/237#issuecomment-203605732

related: #643

### CODE REVIEW

**Comment by lasers** [comment]
With `whatismyip { format = "{country}" }`, I get `"United States"`.
I didn't put quotes in the format. Otherwise, it looks good. 👍

File: py3status/modules/whatismyip.py [hunk 1]
```diff
@@ -58,17 +66,23 @@ def on_click(self, event):
     def _get_my_ip(self):
         """
         """
+        country = ""
         try:
+            if '{country}' in self.format:
```
Comment (tobes) [inline py3status/modules/whatismyip.py]: please use `if self.py3.format_contains(self.format, 'country'):` that way we don't have problems with odd formats like `{country:10}`

File: py3status/modules/whatismyip.py [hunk 2]
```diff
@@ -14,9 +19,11 @@
     timeout: how long before deciding we're offline (default 5)
     url: change IP check url (must output a plain text IP address)
         (default 'http://ultrabug.fr/py3status/whatismyip')
+    url_geo: (default 'http://ip-api.com/csv')
```
Comment (tobes) [inline py3status/modules/whatismyip.py]: should have a description of what option is for

File: py3status/modules/whatismyip.py [hunk 3]
```diff
@@ -4,7 +4,12 @@
 
 Configuration parameters:
     cache_timeout: how often we refresh this module in seconds (default 30)
-    format: what to display (default '{ip}')
+    format: available placeholders are {ip} and {country}
+            If {country} is used the data is queried from
+            "http://ip-api.com/csv"
+            instead of
+            "http://ultrabug.fr/py3status/whatismyip".
```
Comment (tobes) [inline py3status/modules/whatismyip.py]: I'd simplify this as the placeholders are described later

**Comment by vicyap** [comment]
The quotes come from the "csv" returned. I can try to remove quotes if you want

**Comment by lasers** [comment]
Yes, please. It weirds me out to see something in quotes (other than window_title* modules).

**Comment by vicyap** [comment]
so I changed `url_geo` to [http://ip-api.com/json](url). In the process, I was thinking maybe we could just use [http://ip-api.com/json](url) for both `url_geo` and `url` because it provides both ip, country, and more information.

**Comment by tobes** [comment]
I'd get the ip in `_get_my_country()` (maybe rename to match) and then decide which to use based it on if country is in the format use `if self.py3.format_contains(self.format, 'country'):` else just get ip as before

currently you are getting the country even if not needed.

**Comment by vicyap** [comment]
alright, I changed the logic for what you described, should be ready to go

one thing to note: ip-api will ban ip addresses that make more than 150 requests in a minute, so if other modules use it (like weather module), that limit will quickly be reached. Is there a way to integrate getting location info in with py3 to control the requests?

File: py3status/modules/whatismyip.py [hunk 4]
```diff
@@ -55,20 +61,27 @@ def on_click(self, event):
...
-            ip = urlopen(self.url, timeout=self.timeout).read()
-            ip = ip.decode('utf-8')
+            if self.py3.format_contains(self.format, 'country'):
+                resp = urlopen(self.url_geo, timeout=self.timeout).read()
+                resp = json.loads(resp)
+                ip = resp['query']
+                country = resp['country']
+            else:
+                ip = urlopen(self.url, timeout=self.timeout).read()
+                ip = ip.decode('utf-8')
```
Comment (tobes) [inline py3status/modules/whatismyip.py]: you still need to do something like `country = None` here

**Comment by lasers** [comment]
Thank you for removing quotes. 👍

**Comment by ultrabug** [comment]
Thanks a lot guys for pushing back a cool feature.

**Pull request merged**